.github/workflows/longitudinal-benchmark.yml

```diff
@@ -0,0 +1,75 @@
+# Longitudinal Benchmarking 
+# 
+# This workflow will run the benchmarks defined in the environment variable BENCHMARKS.
+# It will collect and aggreate the benchmark output, format it and feed it to github-action-benchmark.
+#
+# The benchmark charts are live at https://input-output-hk.github.io/plutus/dev/bench
+# The benchmark data is available at https://input-output-hk.github.io/plutus/dev/bench/data.js
+
+name: Longitudinal Benchmarking
+
+on:
+  push:
+    branches:
+      - master
+
+permissions:
+  # Deployments permission to deploy GitHub pages website
+  deployments: write
+  # Contents permission to update benchmark contents in gh-pages branch
+  contents: write
+
+jobs:
+  new-benchmark:
+    name: Performance regression check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.3.0
+
+      - name: Run benchmarks
+        env: 
+          BENCHMARKS: "validation validation-decode"
+        run: | 
+          for bench in $BENCHMARKS; do 
+            2>&1 cabal run "$bench" | tee "$bench-output.txt"
+          done 
+          
+          read -r -d '' PYTHON_SCRIPT <<- END_SCRIPT
+              import json 
+              result = []
+              for benchmark in "$BENCHMARKS".split():
+                  with open(f"{benchmark}-output.txt", "r") as file:
```

Comment:
You decided not to use criterion's JSON output? seems like it might have been a bit easier, but not a big deal I guess.